### ISSUE TITLE
Arguments for a call to a spy are not available when spy is called using .apply()

### ISSUE THREAD

**Issue by glosee** [jasmine/jasmine#1479]
## Are you creating an issue in the correct repository?

I think so! I don't think this has anything to do with Karma, as the issue is with jasmine's spies. Forgive me if this should be posted somewhere else. I've tried to search through existing issues and on StackOverflow, but I can't find anything specific to my use case.

## Expected Behavior

Given a function that is a spy
When said function is called using `.apply`
Then I should be able to see the arguments that were passed using `argsFor`

## Current Behavior

The return value of `argsFor()` is an empty array when inspecting a spy after it's been called using `.apply`, instead of the arguments that were passed

## Suite that reproduces the behavior (for bugs)

```javascript
describe('spying and applying', function() {
  it('should store arguments when you use apply', () => {
    const spy = jasmine.createSpy('mySpy');
    spy.apply(['foo', 'bar', 'baz']);
    expect(spy.calls.count()).toEqual(1); // Passes
    expect(spy.calls.argsFor(0)).toEqual('foo', 'bar', 'baz'); // Fails
  });
});
```

## Context
Basically I have a function that accepts both a string or an array, and if it is an array it applies said array to another function (which is spied in my tests). The code looks like this:

```javascript
function trackBeforePushingTo(payload, pushTo, ...pushArgs) {
  if(Array.isArray(payload)) {
    track.apply(payload);
  } else {
    track(payload);
  }
  pushTo(...pushArgs);
}
```

When I test passing an array into this function, the value of `count()` is correct, but the value of `argsFor()` is an empty array.

## Your Environment

* Version used: 2.5.2
* Environment name and version (e.g. Chrome 39, node.js 5.4): Node 6.9.1, PhantomJS (via Karma)
* Operating System and version (desktop or mobile): MacOS Sierra

Thanks!

**Comment by sgravrock** [comment]
It sounds like you're trying to pass `payload` as the arguments to `track`, but your code under "Context" actually passes it as the `this`. Instead of `track.apply(payload)`, you probably want to do something like `track.apply(null, payload)`. Then you'll also want to change the call to `toEqual` to take an array containing all the arguments. Putting it all together, your example test suite should look something like this:

```
describe('spying and applying', function() {
  it('should store arguments when you use apply', () => {
    const spy = jasmine.createSpy('mySpy');
    spy.apply(null, ['foo', 'bar', 'baz']);
    expect(spy.calls.count()).toEqual(1); // Passes
    expect(spy.calls.argsFor(0)).toEqual(['foo', 'bar', 'baz']); // Passes
  });
});
```

If you actually wanted to pass `payload` as `this`, you can test that by checking `spy.calls.first().object`.

**Comment by glosee** [comment]
ah crap... I'm so embarrassed that I opened this ticket. I've used `apply` hundreds of times, but today I somehow forgot that the first argument is the context. I definitely meant to pass `null` as the first argument. Thank you _very_ much for taking the time to reply to my silly mistake. I will close this.

**Comment by sgravrock** [comment]
No worries. We've all gotten `apply` wrong a time or ten.
